core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -0,0 +1,180 @@
+      generator.writeNumberField(CURRENT_SCHEMA_ID, metadata.currentSchemaId());
+    }
+
+    if (metadata.schemas() != null) {
+      JsonUtil.writeObjectList(SCHEMAS, metadata.schemas(), SchemaParser::toJson, generator);
+    }
+
+    generator.writeNumberField(CURRENT_VERSION_ID, metadata.currentVersionId());
+    JsonUtil.writeObjectList(VERSIONS, metadata.versions(), ViewVersionParser::toJson, generator);
+    JsonUtil.writeObjectList(
+        VERSION_LOG, metadata.history(), ViewHistoryEntryParser::toJson, generator);
+
+    generator.writeEndObject();
+  }
+
+  public static ViewMetadata read(InputFile file) {
+    try (InputStream is = file.newStream()) {
+      return fromJson(JsonUtil.mapper().readValue(is, JsonNode.class));
+    } catch (IOException e) {
+      throw new RuntimeIOException(e, "Failed to read file: %s", file);
+    }
+  }
+
+  public static ViewMetadata fromJson(JsonNode node) {
+    return fromJson(node, null);
+  }
+
+  public static ViewMetadata fromJson(JsonNode node, String metadataFileLocation) {
+
+    Preconditions.checkArgument(
+        node.isObject(), "Cannot parse metadata from a non-object: %s", node);
+
+    int formatVersion = JsonUtil.getInt(FORMAT_VERSION, node);
+    Preconditions.checkArgument(
+        formatVersion <= ViewMetadata.SUPPORTED_VIEW_FORMAT_VERSION,
+        "Cannot read unsupported version %s",
+        formatVersion);
+
+    String location = JsonUtil.getString(LOCATION, node);
+
+    int currentVersionId = JsonUtil.getInt(CURRENT_VERSION_ID, node);
+
+    Map<String, String> properties = JsonUtil.getStringMap(PROPERTIES, node);
+
+    List<ViewVersion> versions =
+        JsonUtil.getObjectList(VERSIONS, node, ViewVersionParser::fromJson);
+
+    List<ViewHistoryEntry> history =
+        JsonUtil.getObjectList(VERSION_LOG, node, ViewHistoryEntryParser::fromJson);
+
+    int numVersionsToKeep =
+        PropertyUtil.propertyAsInt(
+            properties,
+            ViewProperties.VERSION_HISTORY_SIZE,
+            ViewProperties.VERSION_HISTORY_SIZE_DEFAULT);
+
+    versions = versions.subList(0, Math.min(numVersionsToKeep, versions.size()));
+
+    List<Schema> schemas = null;
+    Integer currentSchemaId = null;
+    if (node.has(SCHEMAS)) {
+      JsonNode schemaArray = node.get(SCHEMAS);
+      Schema currentSchema = null;
+      Preconditions.checkArgument(
+          schemaArray.isArray(), "Cannot parse schemas from non-array: %s", schemaArray);
+      // current schema ID is required when the schema array is present
+      currentSchemaId = JsonUtil.getInt(CURRENT_SCHEMA_ID, node);
+      // parse the schema array
+      ImmutableList.Builder<Schema> builder = ImmutableList.builder();
+      for (JsonNode schemaNode : schemaArray) {
+        Schema schema = SchemaParser.fromJson(schemaNode);
+        if (schema.schemaId() == currentSchemaId) {
+          currentSchema = schema;
+        }
+        builder.add(schema);
+      }
```

Comment:
The doubt I still have is I think schemas and current schema should always be set, and thus the spec should be reflected so that it's required. Maybe there's a case that we don't want schema for the entire view. But at least in my mind for any SQL representation in the view, the schema-id should be required (which then means there must be a schema list considering we only support SQL representations)